compiler/rustc_target/src/spec/mod.rs

```diff
@@ -1316,6 +1316,32 @@ bitflags::bitflags! {
 rustc_data_structures::external_bitflags_debug! { SanitizerSet }
 
 impl SanitizerSet {
+    const MUTUALLY_EXCLUSIVE: &'static [(SanitizerSet, SanitizerSet)] = &[
+        (SanitizerSet::MEMORY, SanitizerSet::ADDRESS),
+        (SanitizerSet::MEMORY, SanitizerSet::LEAK),
+        (SanitizerSet::THREAD, SanitizerSet::ADDRESS),
+        (SanitizerSet::THREAD, SanitizerSet::LEAK),
+        (SanitizerSet::THREAD, SanitizerSet::MEMORY),
+        (SanitizerSet::HWADDRESS, SanitizerSet::ADDRESS),
+        (SanitizerSet::HWADDRESS, SanitizerSet::MEMORY),
+        (SanitizerSet::HWADDRESS, SanitizerSet::THREAD),
+        (SanitizerSet::MEMTAG, SanitizerSet::ADDRESS),
+        (SanitizerSet::MEMTAG, SanitizerSet::HWADDRESS),
+        (SanitizerSet::KCFI, SanitizerSet::CFI),
+        (SanitizerSet::KERNELADDRESS, SanitizerSet::ADDRESS),
+        (SanitizerSet::KERNELADDRESS, SanitizerSet::LEAK),
+        (SanitizerSet::KERNELADDRESS, SanitizerSet::MEMORY),
+        (SanitizerSet::KERNELADDRESS, SanitizerSet::THREAD),
+        (SanitizerSet::KERNELADDRESS, SanitizerSet::HWADDRESS),
+        (SanitizerSet::KERNELADDRESS, SanitizerSet::MEMTAG),
+        (SanitizerSet::SAFESTACK, SanitizerSet::ADDRESS),
+        (SanitizerSet::SAFESTACK, SanitizerSet::LEAK),
+        (SanitizerSet::SAFESTACK, SanitizerSet::MEMORY),
+        (SanitizerSet::SAFESTACK, SanitizerSet::THREAD),
+        (SanitizerSet::SAFESTACK, SanitizerSet::HWADDRESS),
+        (SanitizerSet::SAFESTACK, SanitizerSet::KERNELADDRESS),
```

Comment:
This doesn't quite seem right. For instance SafeStack is compatible with all sanitizers. In fact, Fuchsia uses it everywhere on X86_64 w/ ASAN. I can't recall if we fully support TSAN or KASAN yet, but it shouldn't be incompatible. 

The canonical definition in clang is here https://github.com/llvm/llvm-project/blob/70b79a9ccd03f93fc4c8464a91b6bef3aab322d3/clang/lib/Driver/SanitizerArgs.cpp#L520

Reply:
The definitions you linked seem to suggest that SafeStack is not compatible with all sanitizers:

```
std::make_pair(SanitizerKind::SafeStack,
               (TC.getTriple().isOSFuchsia() ? SanitizerMask()
                                             : SanitizerKind::Leak) |
                   SanitizerKind::Address | SanitizerKind::HWAddress |
                   SanitizerKind::Thread | SanitizerKind::Memory |
                   SanitizerKind::KernelAddress),
```

Though this change doesn't currently capture that Fuchsia specifically allows LSAN + SafeStack.

Reply:
Hmm, maybe it is ok, I realize now that we disable SafeStack w/ ASAN, so disregard. But the link can be a good cross reference.